tests/basic/volume_ops_test.go

```diff
@@ -3240,3 +3240,144 @@ func writeFioDataToVolume(volName string, n node.Node, size int64) error {
 	return nil
 
 }
+
+var _ = Describe("{OverCommitVolumeTest}", func() {
+	/*
+		    https://portworx.atlassian.net/browse/PTX-19103
+			1. Update the pxctl cluster with cluster option OverCommitPercent with the maximum storage percentage volumes can provision against backing storage set to 100(Enabeling Thick Provisioning)
+		    2. Check the overall storage pool capacity of a particular node
+		    3. create a volume with a base size , for eg: 10 GB
+		    4. Now update the volume size upto the maximum limit of the storage pool eg: if storage pool has a capacity of 200 GB left , increase vol size by 200
+		    5. Check vol size is successful or not
+		    6. Now try to increase the volume size by 1 more GB, it should fail , as it exceeds the storage pool capacity
+			7. Revert back the imposed cluster options
+			8. Create A volume after disabling thick provisioning , just to validate volume creation is happening as expected
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("OverCommitVolumeTest", "Validate Overcommit volume size", nil, 0)
+	})
+	// check the size left in the node
+	itLog := "honor OverCommitPercent when resizing the volume"
+	It(itLog, func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
```

Comment:
add all the use cases mention in QA notes in https://purestorage.atlassian.net/browse/PWX-31840